### DISCUSSION TITLE
[Feature] using multiple level of specificity or reintroducing the glob pattern for resolutions

### DISCUSSION THREAD

**Discussion by eltonio450** [yarnpkg/berry#2846] [Ideas]
Hello everyone,

I digged a bit in the repo to find related issues, but none seems to be about this problem with the "resolutions" property

**Describe the user story**

As a developper, I sometimes would like to overwrite a dependency of a dependency, especially when using workspaces

For example, I want to overwrite the resolution of detective-typecript, by precinct, only in my @main/mypackage workspace

example :  `"@main/myPackage/madge/precinct/detective-typescript": "^7.0.0"`

with yarn 1 and/or without workspace, it was easy

**Describe the solution you'd like**

With yarn 1, it was possible to use the glob pattern  `"@main/myPackage/**/detective-typescript": "^7.0.0"` to make the change specific to a workspace

**Describe the drawbacks of your solution**

I am not sure why this possibility has been removed, but I guess there is a reason, and there might be drawback to multiple level resolutions overwriting

**Describe alternatives you've considered**

yarn patch seems to be an alternative, but makes updating the package a bit trickier I guess

If I missed something, please let me know

Thank you

**Comment by arcanis** [comment]
> I am not sure why this possibility has been removed, but I guess there is a reason

Because it didn't work 🙂 The original syntax was designed before I had enough experience with Yarn, and cannot work given some core assumptions of our architecture (namely, that a same package name/version always has the same dependencies regardless its branch in the dependency tree, only exception being peer deps).
